lib/timef/time_diff.go

```diff
@@ -0,0 +1,96 @@
+// time format, time calculation, all time related things.
+package timef
```

Comment:
I chose the package name `timef` since I didn't want to use the native `time` package name and it is a short for `time_format`